### PR TITLE
WeightOnlyLinear keeps self.weight after recover

### PR DESCRIPTION
**Pull request by xin3he**
## Type of Change

user experience enhancement

## Description

WeightOnlyLinear keeps self.weight after recover due to recover is too slow.

## Expected Behavior & Potential Risk

UT pass

### CODE REVIEW

File: neural_compressor/adaptor/torch_utils/model_wrapper.py [hunk 1]
```diff
@@ -329,6 +326,10 @@ def __init__(
             self.g_idx = None
 
     def pack(self, int_weight, scale, zp, bias, g_idx=None):
+        if self.use_optimum_format:
+            self.scales = self.scales.T
```
Comment (PenghuiCheng) [inline neural_compressor/adaptor/torch_utils/model_wrapper.py:330]: Using in-place mode here, is it right?

Comment (xin3he) [reply neural_compressor/adaptor/torch_utils/model_wrapper.py:330]: `Transpose` in Torch won't introduce a new tensor.

File: neural_compressor/adaptor/torch_utils/model_wrapper.py [hunk 2]
```diff
@@ -468,12 +469,13 @@ def recover(self):
...
-            weight = weight.float()
-            self.bias = self.bias.float() if self.bias is not None else None
-        if level == DEBUG:
+        if not hasattr(self, "weight"):
+            weight = self.recover()
+            device = self.scales.device
+            if weight.dtype == torch.float16 and device.type == "cpu":
+                weight = weight.float()
+                self.bias = self.bias.float() if self.bias is not None else None
+        if True:  # keep reusing self.weight due to recover is too slow.
```
Comment (PenghuiCheng) [inline neural_compressor/adaptor/torch_utils/model_wrapper.py:478]: if it is True, then "else" is not needed.

Comment (xin3he) [reply neural_compressor/adaptor/torch_utils/model_wrapper.py:478]: keep it for debug usage.

**Review by changwangss — APPROVED** [review]

**Review by PenghuiCheng — APPROVED** [review]

**Pull request merged**